build-scripts/relabel_ids.py

```diff
@@ -23,16 +23,22 @@ def parse_args():
     parser = argparse.ArgumentParser(
         description="This script finds check-content files (currently, "
         "OVAL and OCIL) referenced from XCCDF and synchronizes all IDs.")
-    parser.add_argument("xccdf_file")
-    parser.add_argument("new_xccdf_file")
-    parser.add_argument("id_name", help="ID naming scheme")
+    parser.add_argument(
+        "--input-xccdf", required=True, help="Input XCCDF file")
+    parser.add_argument(
+        "--output-xccdf", required=True, help="Output XCCDF file")
```

Comment:
I just saw now that there is a description before this function, informing this script requires two arguments. This is no longer true. The description should be updated or the first sentence could be simply removed:
`This script requires two arguments: an "unlinked" XCCDF file and an ID name scheme.`

Reply:
Late to the party, but perhaps somebody will notice - the convention (also mentioned in the argparse documentation) is that optional arguments (i.e. arguments preceded with -s hort or --long-options) shouldn't be required - the correct way to go is to have multiple named positional arguments.

Reply:
Actually, the documentation says the `required` parameter `should be avoided when possible`. It is not a technical limitation but only likely impacts the user expectation. In practical terms, who would use this script without some of these required options would see an error message informing the required options:
```
$ ./relabel_ids.py 
usage: relabel_ids.py [-h] --input-xccdf INPUT_XCCDF --output-xccdf OUTPUT_XCCDF --output-oval OUTPUT_OVAL --output-ocil OUTPUT_OCIL --id-name ID_NAME
relabel_ids.py: error: the following arguments are required: --input-xccdf, --output-xccdf, --output-oval, --output-ocil, --id-name
```
I believe that in this context mentioned by @jan-cerny , the readability is a good argument to use this parameter in an option considering the impact of a expectation misalignment in this case.